pkg/gotenberg/supervisor.go

```diff
@@ -167,6 +173,11 @@ func (s *processSupervisor) Healthy() bool {
 }
 
 func (s *processSupervisor) Run(ctx context.Context, logger *zap.Logger, task func() error) error {
+	currentQueueSize := s.reqQueueSize.Load()
+	if s.maxQueueSize > 0 && currentQueueSize >= s.maxQueueSize {
```

Comment:
Sorry to wake up this old PR, but I'm wondering if there isn't a potential concurrency issue here.

I'm not a Go expert, but AFAIK atomics only guarantee safe read/write access to a variable, but this is not comparable to a _critical section_ created with a mutex. Here, it means that `10` Goroutines could read the `reqQueueSize` value at the _same_ time, succeed to compare it to `maxQueueSize=5` and then incrementing all together the counter. The increment is safe (because it uses atomic), but it seems to be a gap in the logic because we will end up with a queue with `10` elements, isn't it?

We should allow other Goroutines to read the value only once comparison and potential increment are done IMHO 🤔 
Am I missing something?
Thx for your feedback 🙇

Reply:
Hey thanks for the feedback @b-viguier!

You’re actually right! Did you experience some issues on your side? Also, do you want to try to fix it in a PR with a _mutex_ instead?

Reply:
> Did you experience some issues on your side?

Nope, I tried to send _a lot_ of concurrent requests locally, and never detected any _overflow_... So I guess this is really an edge case

> Also, do you want to try to fix it in a PR with a mutex instead?

Unfortunately, I'm already running out of time, and Go is not my usual dev stack... But if you think it may help, I can open an issue with a clear description of the problem, in case someone else want to give it a try?

BTW: thanks for your great work on this tool 👏🙂

Reply:
No problem! 😉 But if you find time to create an issue, it would be great to help me tracking this edge case! Thanks 😊